labs/b5.md

```diff
@@ -0,0 +1,364 @@
+
+    Moreover ping also provides metrics for Round Trip Time (RTT) and packet loss.  Round trip time is defined as the time it takes for a response to arrive after sending the ping packet.  These can prove to be very useful statistics.
+
+3. `traceroute`
+Traceroute sends packets Time to Live (TTL) equal to the number of hops. Routers decreases the value of TTL of an incoming packet and if it sees an incoming packet with TTL = 0 then drops it, otherwise it decreases the value and sends it further. At the same time it sends diagnosing information to the source about router's identity.
+
+    Traceroute provides a detailed view of the routers that a packet traverses while on its way to a destination.
+
+    If router does not respond within a timeout then traceroute prints an asterisk.
+
+4. `arp`
+Provides info on and the ability to manipulate the ARP cache of the system.
+
+    With arp you can display the system arp table.  Add, remove, or modify arp entries and much more.
+    
+5. `dig`
+Utility for doing dns query and triaging DNS issues.
+
+    Dig by default performs queries to nameservers in `/etc/resolv.conf` but some options allow you to: specify name server, choose query type (iterative vs recursive), and much more -- making `dig` a very flexible DNS tool.
+    
+6. `ip`
+`ip` is a command with many subcommands offering a lot of functionality -- so much it can be overwhelming at first. You will most commonly be using ip to display/modify routing, IP addresses, or network interfaces.
+
+    It will take time to get use to how much functionality is included in this command but for reference here is a pretty compact [cheatsheet].
+
+[cheatsheet]: https://access.redhat.com/sites/default/files/attachments/rh_ip_command_cheatsheet_1214_jcs_print.pdf
+
+7. `netstat`
+This tool is good for printing network connections, routing tables, and probing sockets, amongst other functions.
+
+    `netstat` also has functionality to probe sockets for activity and displays information such protocol (UDP/TCP)
+    
+    If you are investigating sockets `ss` and `lsof` are also options you may want to consider
+8. `tcpdump`
+Perfect for monitoring incoming or outgoing traffic on a machine. 
+
+    `tcpdump` offers countless options when it comes to analyzing traffic: it can capture packets, log traffic, compute metrics, filter traffic, monitor specific interfaces, etc. As a primer you can check out these [examples][tcp].
+
+[tcp]: https://www.rationallyparanoid.com/articles/tcpdump.html
+
+9. `nc`
+A very powerful tool that can be used for just about anything involving TCP or UDP.  It can open TCP connections, send UDP packets, listen on arbitrary TCP and UDP ports, do port scanning, and deal with both IPv4 and IPv6.
+
+10. `curl`
+`curl` is a tool to transfer data from or to a server using certain protocols such as HTTP, FTP, etc ... 
```

Comment:
I'd say something like "cURL does as its name suggests, and allows you to see the contents at certain URLs. Beyond this it's also an extremely powerful program that lets you interact with and inspect servers over several different protocols..."